core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1791,6 +1791,14 @@ private boolean needNewSubQuery(
         return false;
       }
       final Clause maxClause = Collections.max(clauses);
+
+      if (rel instanceof Project
+          && clauses.contains(Clause.ORDER_BY)
+          && dialect.getConformance().isSortByOrdinal()) {
```

Comment:
As mentioned in the Jira, if the sort-by-ordinal is not enabled, the result is also incorrect. Could you also add a test case which shows that your fix also cover it?

Reply:
Thanks for your review,  I add unit test to covert it.